flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SharedSlotTest.java

```diff
@@ -234,6 +236,45 @@ public void testReleaseIfPhysicalSlotIsAllocated() {
 		assertThat(released.isDone(), is(true));
 	}
 
+	@Test
+	public void tesDuplicatedReturnLogicalSlotFails() {
+		CompletableFuture<PhysicalSlot> slotContextFuture = CompletableFuture
+			.completedFuture(new TestingPhysicalSlot(RP, new AllocationID()));
+		AtomicInteger released = new AtomicInteger(0);
+		SharedSlot sharedSlot = SharedSlotBuilder
+			.newBuilder()
+			.withSlotContextFuture(slotContextFuture)
+			.withReleaseCallback(g -> released.incrementAndGet())
```

Comment:
How about adding a case which will invoke `SharedSlot#release()` in the release callback?
This can happen in production code path.

Reply:
I think this case is already covered by `testReleaseEmptyDoesNotCallAllocatorReleaseBack` where `returnLogicalSlot` calls the callback and then next `sharedSlot.release` checks that the callback is not called any more. It is not exactly the same but the infinite recursion cannot happen and the setup will be a bit complicated.